cobalt/dom/source_buffer_metrics.h

```diff
@@ -15,58 +15,63 @@
 #ifndef COBALT_DOM_SOURCE_BUFFER_METRICS_H_
 #define COBALT_DOM_SOURCE_BUFFER_METRICS_H_
 
+#include "base/time/default_tick_clock.h"
+#include "base/time/tick_clock.h"
+#include "base/time/time.h"
 #include "starboard/types.h"
 
 namespace cobalt {
 namespace dom {
 
-#if defined(COBALT_BUILD_TYPE_GOLD)
-
-class SourceBufferMetrics {
- public:
-  explicit SourceBufferMetrics(bool is_primary_video) {}
-  ~SourceBufferMetrics() = default;
-
-  void StartTracking() {}
-  void EndTracking(size_t size_appended) {}
-  void PrintCurrentMetricsAndUpdateAccumulatedMetrics() {}
-  void PrintAccumulatedMetrics() {}
+enum class SourceBufferMetricsAction : uint8_t {
+  PREPARE_APPEND,
+  APPEND_BUFFER,
 };
 
-#else  // defined(COBALT_BUILD_TYPE_GOLD)
-
 class SourceBufferMetrics {
  public:
-  explicit SourceBufferMetrics(bool is_primary_video)
-      : is_primary_video_(is_primary_video) {}
+  SourceBufferMetrics(
+      bool is_primary_video,
+      const base::TickClock* clock = base::DefaultTickClock::GetInstance())
+      : is_primary_video_(is_primary_video), clock_(clock) {}
   ~SourceBufferMetrics() = default;
 
   void StartTracking();
```

Comment:
lets add the SourceBufferMetricsAction enum to StartTracking too, to be consistent with EndTracking. Lets add a DCHECK or similar to ensure the EndTracking matches the correct StartTracking